java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -1168,8 +1193,9 @@ public OrcIndex readRowIndex(int stripeIndex, boolean[] included,
       sargColumns = sargColumns == null ?
           (sargApp == null ? null : sargApp.sargColumns) : sargColumns;
     }
-    return dataReader.readRowIndex(stripe, stripeFooter, included, indexes, sargColumns,
-        bloomFilterIndex);
+    return dataReader.readRowIndex(stripe, evolution.getFileType(0), stripeFooter,
```

Comment:
Actually RecordReaderImpl.mapSargColumnsToOrcInternalColIdx already maps column names to the physical file column ids. It handles ACID files by looking for the field names in row field instead of the root. So at this level, it should be consistently using the file id.